dojo/tools/bugcrowd_api/api_client.py

```diff
@@ -0,0 +1,140 @@
+            else:
+                next = "over"
+                raise Exception(
+                    "Unable to get bugcrowd submissions due to {} - {}".format(
+                        response.status_code, response.content.decode("utf-8")
+                    )
+                )
+
+    def test_connection(self):
+        # Request programs
+        response_programs = self.session.get(
+            url="{}/programs".format(self.bugcrowd_api_url)
+        )
+
+        # Request submissions to validate the org token
+        response_subs = self.session.get(
+            url="{}/submissions".format(self.bugcrowd_api_url)
+        )
+
+        if response_programs.ok and response_subs.ok:
+            data = response_programs.json().get("data")
+            data_subs = response_subs.json().get("meta")
+            total_subs = str(data_subs["total_hits"])
+
+            progs = list(filter(lambda prog: prog["type"] == "program", data))
+            program_names = ", ".join(
+                list(map(lambda p: p["attributes"]["code"], progs))
+            )
+            # Request targets to validate the org token
+            response_targets = self.session.get(
+                url="{}/targets".format(self.bugcrowd_api_url)
+            )
+            if response_targets.ok:
+                data_targets = response_targets.json().get("data")
+                targets = list(
+                    filter(lambda prog: prog["type"] == "target", data_targets)
+                )
+                target_names = ", ".join(
+                    list(map(lambda p: p["attributes"]["name"], targets))
+                )
+                return f'With {total_subs} submissions, you have access to the "{ program_names }" programs, \
+                    you can use these as Service key 1 for filtering submissions \
+                        You also have targets "{ target_names }" that can be used in Service key 2'
+        else:
+            raise Exception(
+                "Connection failed (error: {} - {})".format(
+                    response_subs.status_code, response_subs.content.decode("utf-8")
+                )
+            )
+
+    def test_product_connection(self, api_scan_configuration):
+        submissions = []
+        submission_gen = self.get_findings(
+            api_scan_configuration.service_key_1, api_scan_configuration.service_key_2
+        )
+        for page in submission_gen:
+            submissions = submissions + page
+        submission_number = len(submissions)
+        return f'You have access to "{submission_number}" submissions (no duplicates)\
+            in Bugcrowd in the Program code "{api_scan_configuration.service_key_1}" \
+            and Target "{api_scan_configuration.service_key_2}" (leave service key 2 empty to get all submissions in program)'
+
+    def get_headers(self):
+
+        self.default_headers["Authorization"] = "Token {}".format(self.api_token)
+
+        return self.default_headers
```

Comment:
This should be in the session object.

Reply:
ah yes, it's not in use so I'm deleting that